lychee-lib/src/filter/mod.rs

```diff
@@ -70,6 +70,13 @@ impl Filter {
         }
     }
 
+    #[must_use]
+    /// Whether the host is excluded from checking
+    pub fn is_host_excluded(&self, uri: &Uri) -> bool {
+        // If loopback IPs are excluded, exclude "localhost" as well, which usually maps to a loopback IP
+        self.exclude_loopback_ips && uri.domain().unwrap_or("") == "localhost"
```

Comment:
`domain()` returns an Option right? I think you can avoid the `unwrap` by using structural matching: `&& uri.domain() == Some("localhost")`.

Reply:
Yeah exactly. Not sure whether `unwrap` is more expensive or `Some`? Without `unwrap`, the compile test returned an error and suggested a pretty ugly looking long conversion for "localhost", so I though converting the domain is easier 😄.

Reply:
Changed it now so we see whether it generally works both.